actionpack/test/dispatch/request/jsonapi_params_parsing_test.rb

```diff
@@ -0,0 +1,53 @@
+require 'abstract_unit'
+
+class JsonapiParamsParsingTest < ActionDispatch::IntegrationTest
+  class TestController < ActionController::Base
+    class << self
+      attr_accessor :last_request_parameters
+    end
+
+    def parse
+      self.class.last_request_parameters = request.request_parameters
+      head :ok
+    end
+  end
+
+  def teardown
+    TestController.last_request_parameters = nil
+  end
+
+  test "parses json params for application json" do
+    payload = "{ \"data\": { \"type\": \"posts\", \"attributes\": { \"title\":"\
+              "\"Hello\", \"date\": \"today\"} , \"relationships\": { "\
+              "\"author\": { \"data\": { \"id\": \"2\", \"type\": \"users\" }"\
+              " }, \"comments\": { \"data\": [ { \"type\": \"comments\","\
+              "\"id\": \"3\" }, { \"type\": \"comments\", \"id\": \"4\" } ]"\
+              " }, \"journal\": { \"data\": null } } } }"
+    expected = { "_type" => "posts", "title" => "Hello", "date" => "today",
+                 "author_id" => "2", "author_type" => "User",
+                 "comment_ids" => ["3", "4"], "comment_types" => ["Comment", "Comment"],
+                 "journal_id" => nil }
+    assert_parses(expected, payload, { 'CONTENT_TYPE' => 'application/vnd.api+json' })
```

Comment:
If I recall, the content_type only affects the params parsing. the Accept header is used to choose the Renderer. So, this implementation, to be complete, still requires a JSON API renderer, right? (Which can be added in a subsequent PR, of course)

header-related files I've found
- actionpack/lib/action_dispatch/http/mime_negotiation.rb
- actionpack/lib/action_dispatch/http/mime_type.rb
- actionpack/lib/action_dispatch/http/request.rb
- actionpack/lib/action_dispatch/middleware/public_exceptions.rb
- actionpack/lib/action_dispatch/middleware/show_exceptions.rb